test/index.ts

```diff
@@ -1,26 +1,37 @@
 import { execSync } from 'child_process';
+// TODO import deepEqual from 'deep-is' works for some
+// TODO import { deepEqual } from 'deep-is' works for others
+// TODO require seems to work for all of them
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const deepEqual = require('deep-is');
 
-export type Test = {
+import { jsonStringify } from '../src/lib';
+
+export type Test<Context = any> = {
     name: string;
     skip?: boolean;
     wait?: number;
-    prep?: () => Promise<any>;
-    test?: () => Promise<AzleResult<boolean, string>>;
+    prep?: (context: Context) => Promise<Context | void>;
+    test?: (context: Context) => Promise<AzleResult<boolean, string, Context>>;
 };
 
 // export type Variant<T> = Partial<T>;
 
-export type AzleResult<T, E> = Partial<{
-    Ok: T;
+export type AzleResult<T, E, Context = any> = Partial<{
+    Ok: { passes: T; context?: Context };
```

Comment:
passes should be succeed, also add an error message thing

Reply:
done